fineract-investor/src/main/java/org/apache/fineract/investor/service/ExternalAssetOwnersWriteServiceImpl.java

```diff
@@ -117,7 +118,8 @@ private void validateSettlementDate(ExternalAssetOwnerTransfer externalAssetOwne
     }
 
     private void validateLoanStatus(LoanIdAndExternalIdData loanIdAndExternalIdAndExternalId) {
-        if (Objects.isNull(loanIdAndExternalIdAndExternalId)) {
+        if (Objects.isNull(loanIdAndExternalIdAndExternalId.getLoanId())
```

Comment:
Its a little lie... we will get the same error even the Loan is not even existing...